client/components/search/search-collection.jsx

```diff
@@ -0,0 +1,100 @@
+/**
+* External dependencies
+*/
+import React from 'react';
+
+const Hider = React.createClass( {
+	displayName: 'Hider',
+
+	propTypes: {
+		hide: React.PropTypes.bool,
+	},
+
+	render() {
+		return (
+			<div
+				className={ 'design-assets__group' }
+				style={ this.props.hide ? { display: 'none' } : {} }
+			>
+				{ this.props.children }
+			</div>
+		);
+	}
+} );
+
+const FilterSummary = React.createClass( {
+	getDefaultProps: function() {
+		return {
+			noResultsText: 'No Results Found'
+		};
+	},
+
+	propTypes: {
+		noResultsText: React.PropTypes.string
+	},
+
+	render() {
+		if ( this.props.items.length === 0 ) {
+			return ( <p>{ this.props.noResultsText }</p> );
+		} else {
+			return null;
+		}
+	}
+} );
+
+export default React.createClass( {
+	displayName: 'Collection',
+
+	shouldWeHide: function( example ) {
+		let filter, searchString;
+
+		filter = this.props.filter || '';
+
+		searchString = example.props.searchTerms;
+
+		if ( this.props.component ) {
+			return example.type.displayName.toLowerCase() !== this.props.component.replace( /-([a-z])/g, '$1' );
```

Comment:
What do you say about using this instead of regular expression replacement?
`this.props.component.split( '-' ).join( '' ).toLowerCase()`